guides/keras_nlp/transformer_pretraining.py

```diff
@@ -0,0 +1,481 @@
+text. Our goal is to predict the sentiment of the snippet. A label of 1 indicates
+positive sentiment, and a label of 0 negative sentiment.
+"""
+
+"""
+### Establish a baseline
+
+As a first step, we will establish a baseline of good performance. We don't actually need
+KerasNLP for this, we can just use core Keras layers.
+
+We will build a simple bag-of-words model, where we learn a positive or negative weight
+for each word in our input. A sample's score is simply the sum of all words that are
+present in the sample.
+"""
+
+# This layer will turn our input sentence into a list of 1s and 0s the same size
+# our vocabulary, indicating whether a word is present in absent.
+multi_hot_layer = keras.layers.TextVectorization(
+    max_tokens=4000, output_mode="multi_hot"
+)
+multi_hot_layer.adapt(sst_train_ds.map(lambda x, y: x))
+# We then  learn a linear regression over that layer, and that's our entire
+# baseline model!
+regression_layer = keras.layers.Dense(1, activation="sigmoid")
+
+inputs = keras.Input(shape=(), dtype="string")
+outputs = regression_layer(multi_hot_layer(inputs))
+baseline_model = keras.Model(inputs, outputs)
+baseline_model.compile(loss="binary_crossentropy", metrics=["accuracy"])
+baseline_model.fit(sst_train_ds, validation_data=sst_val_ds, epochs=5)
+
+"""
+A bag-of-words approach can be a fast and suprisingly powerful, especially when input
+examples contain a large number of words. With shorter sequences, it can hit a
+performance ceiling.
+
+To do better, we would like to build a model that can evaluate words *in context*. It's
+more useful to see the phrase "not my cup of tea" in an input, than to be told a sample
+contained the word "cup" or "tea" individually. Instead of evaluating each word in a
+void, we need to use the information contained in the *entire ordered sequence* of our
+input.
+
+This runs us into a problem. `SST-2` is very small dataset, and there's simply not enough
+example text to attempt to build a larger, more parameterized, models that can learn on a
+sequence. We would quickly start to memorize our training set, without any increase in
+our ability to generalize to unseen examples.
+
+Enter **pre-training**, which will allow us to learn on a larger corpus, and transfer our
+knowledge to the `SST-2` task. And enter **KerasNLP**, which will allow us to pre-train a
+particularly powerful model, the transformer, with ease.
+"""
+
+"""
+## Pre-training
+
+To beat our baseline, we need to leverage the `WikiText103` dataset, an unlabeled
+collection of wikipedia articles that is much bigger than than `SST-2`.
+
+We are going to train a *transformer* model, a highly expressive model which will learn
+to embed each word in our input as a low dimentional vector. Our wikipedia dataset has no
+labels, so we will use an unsupervised training objective called the *Masked Language
+Modeling* (MLM) ojective.
+
+Essentially, we will be playing a big game of "guess the missing word." For each input
```

Comment:
nit: move period out of "".

I am not actually sure about this though...

Reply:
Yeah, probably debatable, but it's a consistent style for this guide at least. Will do.

For technical writing, I think the "quote." style is terrible.